usl_models/usl_models/flood_ml/model.py

```diff
@@ -325,47 +278,54 @@ def __init__(
             name="output_cnn",
         )
 
-    def set_n_predictions(self, n_predictions: int) -> None:
-        """Updates n_predictions."""
-        self._n_predictions = n_predictions
-
-    def forward(
-        self,
-        st_input: st_tensor,
-        geo_input: geo_tensor,
-        temp_input: temp_tensor,
-    ) -> tf.Tensor:
+    def call(self, input: Input) -> tf.Tensor:
         """Makes a single forward pass on a batch of data.
 
         The forward pass represents a single prediction on an input batch
         (i.e., a single flood map). This functions implements the logic of the
         internal ConvLSTM and ignores autoregressive steps.
 
         Args:
-            st_input: Flood maps tensor of shape [B, n, H, W, 1].
-            geo_input: Geospatial tensor of shape [B, H, W, f].
-            temp_input: Rainfall windows tensor of shape [B, n, m].
+            input: Dictionary containing: 
+              - spatiotemporal: Flood maps tensor of shape [B, n, H, W, 1].
+              - geospatial: Geospatial tensor of shape [B, H, W, f].
+              - temporal: Rainfall windows tensor of shape [B, n, m].
 
         Returns:
             The flood map prediction. A tensor of shape [B, H, W, 1].
         """
+        spatiotemporal: tf.Tensor = input["spatiotemporal"]
+        geospatial: tf.Tensor = input["geospatial"]
+        temporal: tf.Tensor = input["temporal"]
+
+        B = spatiotemporal.shape[0]
+        C = 1
```

Comment:
Can we maybe add a comment saying this is the channel dimension for the spatiotemporal tensors? Unlike the other variables, there's no context to infer the variable meaning.

Reply:
Done.